docs/next_link_analysis.md

```diff
@@ -0,0 +1,108 @@
+
+## Illustration of data
+
+![visitor data](./images/visit_data_example.jpeg)
+
+As seen, pagePath represents the URL visited.
+
+## Rationale for considering only the preceding website
+
+Typically in time series processes, the most important data point is the current one, in determining the next data point. In a memoryless process, the only data point which is relevant is the current one and the following data point will be a random jump to one of the connected points in the space. In the context of the GovUK website, it is frequently the case that a visitor can only visit the next URL via the current URL.
+
+### Conditioning on time
+
+As we are focussing on the single hour of data, we have implicitly conditioned on time. Different  predictions would be made for different slices of time. This would mean that this approach could incoporate non-stationary effects and the impact of current events. As will be discussed beneath, the model could be run in real time, as it is effectively relying on aggregations and dictionary data access which is inherently fast.
+
+### Conditioning on location
+
+Location is not considered here, but would be possible given that there is information about geolocation of the user. This simply requires consideration of aggregation weights as a function of distance, discounting information from users further away.
+
+## Preparing the lagged data
+
+The data requires a lagged URL to be created in order to see the link between current and past data. We are trying to predict the URL to be visited, by referencing the lagged URL.
+
+Of note, a full 53% of the users on the government website only visit one link. This makes it impossible to create a lagged URL. These users data are removed from the dataset, as we cannot build a prediction conditioned on the past, as there is no history for the user.
+
+### Removing login, search and done activity
+
+Any URLs for search or login are removed. Search activity only accounts for [4%] of the remaining data, which when considered in combination with the number of users who only visit one URL, is indicative of external search engines.
+
+Login activity, [6%] of the data, was removed as its a deterministic intermediate link to certain URLs, ie the probability of `P(a->login->b)` is the same as `P(a->b)`, as `P(login->b)` is 100%. Which in turn would mean that it obscures the relationship `P(a -> b)`. This admittedly ignores the losses from people who forget their login details.
+
+Done activity was also removed, approximately a further [1%] of the data. The rationale for removing done activity was that it just represents the completion of an activity or form and is not helping the user access information.
```

Comment:
Is it also a completely deterministic next step. e.g. One is going through a page on *how to do a thing*, stepping through the instructions and clicking next, then they reach the end and get a *Done* page. In which case, we wouldn't want to try to predict this, right?

Reply:
I think so, at least, I think it makes sense to look at where the prediction errors are occurring and focus on those. There should be not prediction errors for the data where someone is filling in a form, going through a deterministic set of connected pages.

Reply:
Yeah, though I think from previous analysis done on this by others, there were instances of user-journeys coming off the deterministic set of pages... 🤯

Reply:
Yes, in the case of form filling, we will have captured any unusual jumps out of the form and into a new parent URL with the way the data has been built...we are only removing the login page.